### ISSUE TITLE
Add "Cancel" option when downloading updates

### ISSUE THREAD

**Issue by cdwhitfield73** [unfoldingWord-dev/ts-desktop#898]
Make sure these boxes are checked before submitting your issue -- thank you!

- [ ] Check that you are using the [latest version](https://github.com/unfoldingWord-dev/ts-desktop/releases/latest) of the app.
- [ ] Check that a similar issue has not already been opened.

**Comment by cdwhitfield73** [comment]
Look into how to make this functionality work.

**Comment by cckozie** [comment]
I think the CANCEL during the update process is also going to cause some user freak-outs. On my "field-like" Windows machine, when I click CANCEL in the update process, the screen goes "blank" for 12 seconds, then for another 23 seconds it shows the home screen with no projects visible. At this point most users will most likely think they have lost all of their projects.
Would it be possible to set a flag in local or session storage when executing the CANCEL, and then when the page reloads check the flag and display some sort of "working on it" dialog until we've gotten through all of the initialization steps?

**Comment by cdwhitfield73** [comment]
Now, after clicking cancel, the main window is hidden while reloading and the splash screen displays (like at startup) until the reload is complete.

**Comment by cckozie** [comment]
It's way better now that it doesn't look like the projects have been deleted! It still might scare the user the first time they see it, but I don't know if there's an alternative.

**Comment by cdwhitfield73** [comment]
The alternative would be to have actual "cancelable" Promise functions, so the cancel button could just "kill" the download and not have to reload the entire app.  This means either waiting for the ES7 spec to come out or trying to incorporate the "bluebird" library module (which claims to have this feature).  For now, if we are going to have to "reload" the entire app on cancel, I think this is the best user experience we can give them.

**Comment by cckozie** [comment]
Is it possible to also hide the splash screen as well and put up a different dialog during that time? Something like "Updates being cancelled. Please wait..."

**Comment by cdwhitfield73** [comment]
I've hidden the main splash screen elements, including the green status bar and set the text as "Canceling Operation.  Please wait..."  This wording is universal whether you are canceling from a "download" or an "update".

**Comment by cckozie** [comment]
Per CJ, "Canceling. Please wait..."

**Comment by cckozie** [comment]
The splash screen appears without any text, and the status messages are being displayed. This is also the case when opening the app.

**Comment by cdwhitfield73** [comment]
After the last build, I had decreased the "delay" between the code that creates the splash screen window and the code that populates the text.  Apparently, I set it to too small of a delay and the window isn't built yet before the text tries to populate.  I have increased the delay again and it works fine on my computer now.  There is a possibility that on really slow systems, it may still show up blank.  If so, the only option is to create 3 separate splash screens instead of using the same element with different text for main loading, academy loading, and re-loading.
